explore/src/main/scala/explore/cache/ProgramCacheController.scala

```diff
@@ -22,198 +18,103 @@ import explore.model.ObservationExecutionMap
+  val modState     = modProgramSummaries
+  // given StreamingClient[IO, ObservationDB] = client
+  given Logger[IO] = logger
 
 object ProgramCacheController
     extends CacheControllerComponent[ProgramSummaries, ProgramCacheController]
     with CacheModifierUpdaters:
 
-  private def drain[A, Id, R](
-    fetch:      Option[Id] => IO[R],
-    getList:    R => List[A],
-    getHasMore: R => Boolean,
-    getId:      A => Id
-  ): IO[List[A]] = {
-    def go(id: Option[Id], accum: List[A]): IO[List[A]] =
-      fetch(id).flatMap(result =>
-        val list = getList(result)
-        if (getHasMore(result)) go(list.lastOption.map(getId), list)
-        // Fetching with offset includes the offset, so .dropRight(1) ensures we don't include it twice.
-        else (accum.dropRight(1) ++ list).pure[IO]
-      )
-
-    go(none, List.empty)
-  }
-
   private def updateProgramTimes(
     programId: Program.Id
-  )(using StreamingClient[IO, ObservationDB]): IO[ProgramSummaries => ProgramSummaries] =
-    ProgramSummaryQueriesGQL
-      .ProgramTimesQuery[IO]
-      .query(programId)
-      .raiseGraphQLErrors
+  )(using odbApi: OdbProgramApi[IO]): IO[ProgramSummaries => ProgramSummaries] =
+    odbApi
+      .programTimes(programId)
       .map:
-        _.program.fold(identity[ProgramSummaries]): times =>
+        _.fold(identity[ProgramSummaries]): times =>
           ProgramSummaries.programTimesPot.replace(Pot(times))
 
   private def updateObservationExecution(
     obsId: Observation.Id
-  )(using StreamingClient[IO, ObservationDB]): IO[ProgramSummaries => ProgramSummaries] =
-    ProgramSummaryQueriesGQL
-      .ObservationExecutionQuery[IO]
-      .query(obsId)
-      // This no longer raises an error on no data, rather the error is put in the pot.
-      .map(_.result match {
-        case Ior.Right(r)   => r.asRight
-        case Ior.Both(_, r) => r.asRight
-        case Ior.Left(e)    => ResponseException(e, none).asLeft
-      })
+  )(using odbApi: OdbObservationApi[IO]): IO[ProgramSummaries => ProgramSummaries] =
+    odbApi
+      .observationExecution(obsId)
       .map:
-        case Left(e)     =>
-          val t = new Exception("Error getting observation execution information.", e)
-          ProgramSummaries.obsExecutionPots.modify(_.setError(obsId, t))
-        case Right(data) =>
-          data.observation
-            .map(_.execution)
-            .fold(identity[ProgramSummaries]): execution =>
-              ProgramSummaries.obsExecutionPots.modify(_.updated(obsId, execution))
+        _.fold(identity[ProgramSummaries]): execution =>
+          ProgramSummaries.obsExecutionPots.modify(_.updated(obsId, execution))
+      .handleError: e =>
+        val t = new Exception("Error getting observation execution information.", e)
+        ProgramSummaries.obsExecutionPots.modify(_.setError(obsId, t))
 
   private def updateGroupTimeRange(groupId: Group.Id)(using
-    StreamingClient[IO, ObservationDB]
```

Comment:
I hate to say it, but this code is not the current `main`. It is prior to my last PR.

Reply:
Well, actually, my second to last PR. There were substantial changes to this file...